tinygrad/codegen/linearizer.py

```diff
@@ -598,10 +598,12 @@ def limit_global_dims(self, limit):
     # sometimes, there's more dimensions than len(self.lang.gid).
     # compact all the dimensions into the first
     # NOTE: this might make multiview shapetrackers
-    if limit and (self.first_reduce-self.local_dims) > limit:
+    if (self.first_reduce-self.local_dims) > limit:
       num_to_merge = ((self.first_reduce-self.local_dims) - limit)+1
       self.reshape_and_permute(lambda x: (prod(x[0:num_to_merge]),)+x[num_to_merge:], None)
-      if DEBUG >= 3: print("reshaped to", self.full_shape, "due to too many global dimensions")
+    global_dims = self.first_reduce-self.local_dims
+    self.reshape_and_permute(lambda x: x[0:global_dims][::-1]+x[global_dims:], None)
```

Comment:
This doesn't seem right. What is it doing?

Reply:
It's forcing the largest allocation to always be aligned with the `global.x` dim in cuda. It was also mis-aligned in opencl but I think it handles device limit internally.

Reply:
I don't think it does that...it just flips the shape. Why is that one the largest?

Reply:
When it is large enough to exceed device limit, due to the flip from [render_kernel](https://github.com/tinygrad/tinygrad/blob/d2c1e8409abb281493e7aa5f7a8efb9055f6a1cf/tinygrad/codegen/cstyle.py#L85), it created  `global_size` such as `[2, 2, 1048576]`, `[16, 65536]` (later converted to `[16, 65536, 1]` by [ops](https://github.com/tinygrad/tinygrad/blob/d2c1e8409abb281493e7aa5f7a8efb9055f6a1cf/tinygrad/ops.py#L142)) which made those allocation to be aligned with `global.y` or `global.z`.